url026791d51b_dhleong_SeriesTracker_v6_tgz-pJ8-stv6_SeriesTrackerJ8/src/stv6/STHandlerManager.java

```diff
@@ -0,0 +1,84 @@
+package stv6;
+
+import stv6.http.request.Request;
+import stv6.http.request.RequestHandlerManager;
+import stv6.http.request.Response;
+
+public class STHandlerManager extends RequestHandlerManager {
+
+	private static final STHandlerManager instance_ = new STHandlerManager();
+	
+	private volatile Boolean isStatic = Boolean.TRUE;
+
+	public static STHandlerManager getInstance() {
+		return instance_;
+	}
+	
+	public Response handle(Request r) {		
+		// first of all, file requests or plugins should always be okay
+		// plugins allowed basically so I can reset mediatomb if it died
+		if (r.getPage().equals(STServer.FILES) ||
+				r.getPage().equals(STServer.PLUGINS))
+			return super.handle(r);
+		 
+		Response resp = new Response();
+		
+		// set the content type for PS3 to be happy
+		// it's not a file, so it should be html!
+		resp.setContentType("text/html"); 
+			 
+		// make sure we have a profile!
+		if (!Profile.getInstance().isSelected() && 
+				specificHandler(STServer.PROFILE_SELECT, r, "display Profile Select page", resp)) {
+			return resp;
+		}
+		
+		// check for a static message (eg: reloading)
+		synchronized(isStatic) {
+			if (isStatic && specificHandler(STServer.STATIC_MESSAGE, r, 
+					"display Static Message page", resp)) {
+				return resp;
+			} 
+		}
+		// make sure we know the user		
+		if (!Profile.getInstance().fillUser(r) &&
+				specificHandler(STServer.USER_SELECT, r, "request a user", resp)) {			
+			return resp;
+		}
+		
+		// default actions
+		return super.handle(r);
+	}
+	
+	/**
+	 * Sets the server to only display the StaticMessageHandler (or not)
+	 * 
+	 * @param isOn
+	 */
+	public void setStaticMessage(boolean isOn) {
+		synchronized(isStatic) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Object isStatic synchronized on line 59 should not be re-assigned in the same synchronized block on line 60. After this re-assignment, you lose the reference to the synchronized object.